projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -85,6 +87,41 @@ public static function build_view_data() {
 			}
 		}
 
+		if ( $is_archive ) {
+			if ( $wp_the_query->is_home() ) {
+				$view_data['home'] = '1';
+			} elseif ( $wp_the_query->is_archive() ) {
+				if ( $wp_the_query->is_date ) {
+					$query               = $wp_the_query->query;
+					$date_parts          = array_filter( array( $query['year'] ?? null, $query['monthnum'] ?? null, $query['day'] ?? null ) );
+					$date                = implode( '/', $date_parts );
+					$view_data['arch']   = 'date';
+					$view_data['arch_v'] = $date;
+				} elseif ( $wp_the_query->is_category ) {
+					$view_data['arch']   = 'cat';
+					$view_data['arch_v'] = $wp_the_query->query['category_name'];
+				} elseif ( $wp_the_query->is_tag ) {
+					$view_data['arch']   = 'tag';
+					$view_data['arch_v'] = $wp_the_query->query['tag'];
+				} elseif ( $wp_the_query->is_author ) {
+					$view_data['arch']   = 'author';
+					$view_data['arch_v'] = $wp_the_query->query['author_name'];
+				} elseif ( $wp_the_query->is_tax ) {
+					$query = $wp_the_query->query;
+					if ( is_array( $query ) && count( $query ) === 1 ) {
+						$view_data['arch']   = array_keys( $query )[0];
+						$view_data['arch_v'] = array_values( $query )[0];
+					}
+				}
+			} elseif ( $wp_the_query->is_404() ) {
+				$view_data['arch']   = 'err';
+				$view_data['arch_v'] = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ?? '' ) ); // send the url path
+			} elseif ( $wp_the_query->is_search() ) {
+				$view_data['arch']   = 'search';
+				$view_data['arch_v'] = $wp_the_query->query['s'];
```

Comment:
My impression is that other values like `category_name`, `tag`, and `author_name` are safe, given they come from WordPress itself. If they are modified in the URL, for instance, that would result in a `404` error.

That said, `s` is a user-supplied value. Would it make sense to use `sanitize_text_field()` here as well?

Reply:
This makes sense! I'll add it in my next commit, along with the taxonomies test.